### PR TITLE
KEYCLOAK-7705 Added hardcoded-ldap-group-mapper for user federation

### PR DESCRIPTION
**Pull request by jloupdef**
Enable the automatic addition of an hardcoded group (given by its name) when federating users from LDAP

### CODE REVIEW

File: federation/ldap/src/main/java/org/keycloak/storage/ldap/mappers/HardcodedLDAPGroupStorageMapperFactory.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * @author <a href="mailto:jean-loup.maillet@yesitis.fr">Jean-Loup Maillet</a>
+ */
+public class HardcodedLDAPGroupStorageMapperFactory extends AbstractLDAPStorageMapperFactory {
+
+    public static final String PROVIDER_ID = "hardcoded-ldap-group-mapper";
+    protected static final List<ProviderConfigProperty> configProperties = new ArrayList<ProviderConfigProperty>();
+
+    static {
+        ProviderConfigProperty groupAttr = createConfigProperty(HardcodedLDAPGroupStorageMapper.GROUP, "Group",
+                "Group to add the user in.  Click 'Select Group' button to browse groups, or just type it in the textbox.",
```
Comment (mposolda) [inline federation/ldap/src/main/java/org/keycloak/storage/ldap/mappers/HardcodedLDAPGroupStorageMapperFactory.java]: I think this tooltip, which will be shown in the admin console, is not correct. The type of the configuration property is String, so there won'be anything like "Select Group" combobox if I am correct? User will need to type the group manually including group path. Could you change the tooltip to something like: 
``
Group to add the user in.  Fill the full path of the group including path. For example '/root-group/child-group' . 
``
WDYT?

If you want, you can contribute "Group Selector" angular component to our admin console. Something like "Role Selector" is currently doing. But that would be separate PR (and separate JIRA) if you want to do this :)

**Review by mposolda — CHANGES_REQUESTED** [review]
Thanks a lot for the PR! I left one comment for change request.

File: federation/ldap/src/main/java/org/keycloak/storage/ldap/mappers/HardcodedLDAPGroupStorageMapperFactory.java (hunk 1 above)
Comment (jloupdef) [reply federation/ldap/src/main/java/org/keycloak/storage/ldap/mappers/HardcodedLDAPGroupStorageMapperFactory.java]: Yes, you are perfectly right.
I wanted to do the group selection component but postponed it... Did not think of correcting the tooltip.

**Comment by mposolda** [comment]
LDAPProvidersIntegrationNoImportTest is still failing.

**Comment by jloupdef** [comment]
Unit tests corrected. Sorry for that.

**Review by mposolda — APPROVED** [review]

**Comment by mposolda** [comment]
@jloupdef Thanks

**Pull request merged**